File: ListViewPaginationAndroid/app/src/main/java/tr/com/fkocak/listviewpaginationandroid/api/RetrofitClient.java
package tr.com.fkocak.listviewpaginationandroid.api;

import retrofit2.Retrofit;
import tr.gov.acsgb.isgkatip.api.convertor.CustomRetrofitConverterFactory;

/**
 * Created by fatihkocak on 25.04.2019.
 */

public class RetrofitClient {
    private static Retrofit retrofit = null;

    public static Retrofit getClient(String baseUrl) {
        if (retrofit==null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(baseUrl)
                    .addConverterFactory(CustomRetrofitConverterFactory.create())
                    .build();

        }
        return retrofit;
    }
}

File: OnedioAndroid/app/src/main/java/com/onedio/androidside/api/RetrofitClient.kt
package com.onedio.androidside.api

import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.converter.scalars.ScalarsConverterFactory
import java.util.concurrent.TimeUnit


class RetrofitClient {
    companion object {
        val retrofit: Retrofit? = null
        /*fun getClient(baseURL : String): Retrofit = if(retrofit == null)
            Retrofit.Builder()
                .baseUrl(baseURL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        else
            retrofit*/

        fun getClient(baseURL: String): Retrofit {

            val okHttpClient = OkHttpClient.Builder()
                .readTimeout(15000, TimeUnit.MILLISECONDS)
                .writeTimeout(15000, TimeUnit.MILLISECONDS)
                .connectTimeout(15000, TimeUnit.MILLISECONDS)
                .build()

            return Retrofit.Builder()
                .baseUrl(baseURL)
                .client(okHttpClient)
                .addConverterFactory(ScalarsConverterFactory.create())
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        }
    }
}